policykit/policyengine/tests/test_exec_code.py

```diff
@@ -40,6 +57,28 @@ def test(x, name="alice"):
             execute_user_code(example, "test", 5)
         self.assertTrue("Import statements are not allowed" in str(cm.exception))
 
+    def test_execute_unsafe_access(self):
+        """Test that getattr is restricted"""
+        example = """
+def test(inst):
+    return inst._private_attr
```

Comment:
The `safer_getattr` guard restricts users from accessing attributes that are prefixed with `_`. This is helpful because we can use this prefix to restrict any functions on communities that we don't want to expose to policy authors (I've already been doing this by convention, but now it's actually enforced). @amyxzhang

Reply:
Awesome!!

Reply:
And these attributes won't show up in the autocomplete as well, right?

Reply:
Yes they don't show up in autocomplete, because we filter them out [here in the autocomplete module](https://github.com/amyxzhang/policykit/blob/6be3d0da3100c104e6d72d646b6b9ecd4c405760/policykit/policyengine/autocomplete.py#L58)